### PR TITLE
Feature/data explorer pathways dropdowns dependency

### PR DESCRIPTION
**Pull request by weberjavi**
This PR displays dropdown options depending on previous selections on Pathways tab of Data Explorer.
[Pivotal](https://www.pivotaltracker.com/story/show/159405550)
Test it on: http://localhost:3000/data-explorer/emission-pathways?page=1

kapture 2018-08-07 at 12 53 23

### CODE REVIEW

**Comment by Bluesmile82** [comment]
There is a crash when we select a model in Pathways and then a location that doesn't have that model. Try: http://localhost:3000/pathways/models?category=10&indicator=388&model=3
Also, check the conflicts, I think is because of the migration.

**Comment by weberjavi** [comment]
☝️ ready 
@j8seangel or @aabdaab please check that out.
The requirements for this task where kind of vague (just _display dropdown options depending on previous selections_), so I have to decide along the way. The intended functionality is:
 - You can freely filter by `location`, `model` and `category`
 - `scenarios`, `subcategories` and `indicators` are disabled until some of the previous dropdowns are selected (as `scenarios` are dependent of a `model` and `subcategories` and `indicators` are dependent of a `category`) 
-  Did not add `location` dependency in any selector because the user could want to filter by `model` or `category` without taking into account the `location`
-  Once a `model` or `category` (or any of their dependents) is chosen if the user selects a `location` probably the filter will end up with _no data_ and user will be forced to clear the filtering to be able to proceed with a successful search.

Please share your thoughts.
@simaob you are also welcome to opine.

🙏

File: app/javascript/app/components/data-explorer-content/data-explorer-content-selectors.js [hunk 1]
```diff
@@ -220,32 +224,111 @@ export const getLink = createSelector(
...
+function getPathwaysCategoryOptions(query, filtersMeta) {
+  if (!query || !query.scenarios) {
+    return filtersMeta.categories.filter(c => c.parent_id === null);
+  }
+  const selectedScenarioId = query.scenarios;
+  const scenarioIndicatorsIds = filtersMeta.scenarios.find(
+    sc => sc.id === selectedScenarioId
+  ).indicator_ids;
+  const categories = [];
+  scenarioIndicatorsIds.forEach(indId =>
```
Comment (aabdaab) [inline app/javascript/app/components/data-explorer-content/data-explorer-content-selectors.js]: I thi

**Comment by weberjavi** [comment]
@aabdaab yeah, I agree about functions being quite big. 
I've been splitting them along the way (the ones I had to deal with), and hope is a job that we could improve on the remaining tasks of the data-explorer.
Maybe is even desirable to move to another folder the ones that do not affect every section of the explorer (as `getPathways...Options` ones)

File: app/javascript/app/components/data-explorer-content/data-explorer-content-selectors.js (hunk 1 above)
Comment (weberjavi) [reply app/javascript/app/components/data-explorer-content/data-explorer-content-selectors.js]: ⚡️ done 
Thanks, much cleaner that way (also spotted a bug 🐛 ⚔️ )

**Comment by j8seangel** [comment]
👍

**Pull request merged**